src/components/connection_handler/include/connection_handler/connection_handler_impl.h

```diff
@@ -716,6 +734,11 @@ class ConnectionHandlerImpl
    */
   Connection* ending_connection_;
 
+  /**
+   * @brief List connection, which processed final message
+   */
+  std::set<transport_manager::ConnectionUID> connections_final_message;
```

Comment:
It seems to me that this map could be replaced with a new bool property for the Connection class `final_message_sent()`. This would eliminate the need for the `ClosureReasonHandling` function

Reply:
@jacobkeeler 
Unfortunately, we may have multiple applications on different connections.
We should be able to identify the connection with "transport_manager :: ConnectionUID".

Reply:
`connection_list_` appears to be a 1-1 map from `ConnectionUID` to `Connection`. Are you telling me that a `Connection` can have multiple `ConnectionUID`s?

Reply:
The main issue I have with this new structure is that there seems to be no guarantee that the connection IDs will actually be removed from this set during closure.

Reply:
@jacobkeeler 
Sorry, small misunderstanding. 
I thought you want change connections_final_message to new bool property in the connection_handler_impl.h.
Now I understand what you mean, sorry.

Reply:
Ah, I also realized from your comment that there were still issues with my solution, perhaps the field should be added to the `Session` class instead, since that would relate to a single app.

Reply:
Adding additional boolean property to the interface saems to me as more complicated and less obvious soultion then set of connection id's with connections_final_messages. 
Also it iwll provide access to this information to all components that wotk with Connection/Session interfaces. For better incapsulation I would leave this solution.

Reply:
I disagree, using a property seems far more appropriate in this case.

> The main issue I have with this new structure is that there seems to be no guarantee that the connection IDs will actually be removed from this set during closure.

Using a property will guarantee that this information will not linger past the lifespan of the `Session` object

Reply:
@LuxoftAKutsan see https://github.com/smartdevicelink/sdl_core/pull/3553 for the changes I'm referring to